crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -179,76 +179,85 @@ def transition(inst, name, state)
   end
 
   def load_repository_bag
-    bag = Chef::DataBag.load("repositories") rescue nil
+    bag = Chef::DataBag.load("crowbar/repositories") rescue nil
     if bag.nil?
-      create_repository_bag
+      @logger.debug("Creating repositories data bag")
+      bag = Chef::DataBagItem.new
+      bag.data_bag "crowbar"
+      bag["id"] = "repositories"
+      bag.save
     end
     bag
   end
 
   def synchronize_repositories(platforms)
-    # create DataBag if it doesn't exist yet
-    load_repository_bag
+    bag = load_repository_bag
+
     platforms.each do |platform, repos|
       repos.each do |reponame, active|
         repo_object = Crowbar::Repository.where(platform: platform, repo: reponame).first
         case active.to_i
         when 0
-          db = Chef::DataBagItem.load("repositories", repo_object.id) rescue nil
-          # skip repo if it hasn't been active before
-          next if db.nil?
-          destroy_repository_item(repo_object)
+          disable_repository(platform, repo, bag)
         when 1
-          create_repository_item(repo_object)
+          enable_repository(platform, repo, bag)
         end
       end
     end
   end
 
-  def create_repository_item(repo_object)
-    saved = false
-    # create DataBag if it doesn't exist yet
-    load_repository_bag
-    # create DataBagItem with the repo metadata used by provisioner recipes
-    if repo_object.available? && repo_object.valid_key_file?
-      @logger.debug("Setting #{repo_object.registry['name']} repository for #{repo_object.platform} as active.")
-      repo_object.to_databag!.save
-      saved = true
-    else
-      @logger.debug("Cannot set #{repo_object.registry['name']} repository for #{repo_object.platform} as active.")
+  def enable_repository(platform, repo, bag = nil)
+    bag = load_repository_bag if bag.nil?
+
+    repo_object = Crowbar::Repository.where(platform: params[:platform], repo: params[:repo]).first
```

Comment:
the params Hash only works in a Controller, probably copy paste error. should be just platform and repo